Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp

```diff
@@ -82,205 +90,153 @@ plTextFont::~plTextFont()
 
 uint16_t  *plTextFont::IInitFontTexture()
 {
-#ifdef HS_BUILD_FOR_WIN32
-    int     nHeight, x, y, c;
+    int     x, y, c;
     char    myChar[ 2 ] = "x";
-    uint16_t  *tBits;
 
-    DWORD       *bitmapBits;
-    BITMAPINFO  bmi;
+    FT_Library  library;
+    FT_Face     face;
+    FT_Error ftError = FT_Init_FreeType(&library);
+    hsAssert(ftError == FT_Err_Ok, "FreeType did not initialize");
+
+#ifdef HS_BUILD_FOR_WIN32
+    
     HDC         hDC;
-    HBITMAP     hBitmap;
     HFONT       hFont;
-    SIZE        size;
     BYTE        bAlpha;
-
-    
-    // Figure out our texture size
-    if( fSize > 40 )
-        fTextureWidth = fTextureHeight = 1024;
-    else if( fSize > 20 )
-        fTextureWidth = fTextureHeight = 512;
-    else
-        fTextureWidth = fTextureHeight = 256;
-
-
-    // Create a new DC and bitmap that we can draw characters to
-    memset( &bmi.bmiHeader, 0, sizeof( BITMAPINFOHEADER ) );
-    bmi.bmiHeader.biSize = sizeof( BITMAPINFOHEADER );
-    bmi.bmiHeader.biWidth = fTextureWidth;
-    bmi.bmiHeader.biHeight = -(int)fTextureHeight;
-    bmi.bmiHeader.biPlanes = 1;
-    bmi.bmiHeader.biCompression = BI_RGB;
-    bmi.bmiHeader.biBitCount = 32;
     
     hDC = CreateCompatibleDC(nullptr);
-    hBitmap = CreateDIBSection(hDC, &bmi, DIB_RGB_COLORS, (void **)&bitmapBits, nullptr, 0);
     SetMapMode( hDC, MM_TEXT );
+    // Get the font data
 
-    nHeight = -MulDiv( fSize, GetDeviceCaps( hDC, LOGPIXELSY ), 72 );
-    fFontHeight = -nHeight;
-
+    int nHeight = -MulDiv( fSize, plWinDpi::Instance().GetDpi(), 72);
```

Comment:
Is this ever re-done when the DPI changes while the game is running?

Reply:
DPI change is treated as a device reset IIRC just like any other resolution change - which should flush textures like this one.

I don't know that this path has ever been fully tested however.